src/document_models.py

```diff
@@ -32,8 +46,111 @@ class Person(Document):
     preferredName = StringField(max_length=240)
     services = DictField()
 
-    # TODO:
+    # TODO: other fields
     # add role at Olin
     # BOW students?
     # allergies/diet
-    # image/gravatar
\ No newline at end of file
+    # image/gravatar
+
+class Token(Document):
+    """ Represents an authorization token that allows a user to delegate a client
+    application access to the Olin API on their behalf. """
+    email = EmailField(max_length=100, required=True, unique=True)
+    # 'validated' is marked as true once the user has confirmed they own the
+    # email address the validation token is sent to
+    validated = BooleanField(default=False)
+
+    def generate_token(self, expiration=86400 * 365):
+        """ Generate an authentication token, by default good for 1 year """
+        serializer = Serializer(
+            current_app.config['SECRET_KEY'], expires_in=expiration)
+        return serializer.dumps({'email': self.email}).decode(encoding='UTF-8')
+
+    @staticmethod
+    def verify_token(token):
+        """ ensures that the login token passed to it is valid, returns
+        the correct person object based on the token passed to it """
+        serializer = Serializer(current_app.config['SECRET_KEY'])
+        try:
+            data = serializer.loads(token)
+        except SignatureExpired:
+            return None # valid token, but expired
+        except BadSignature:
+            return None # invalid token
+        token = Token.objects.get(email=data['email'])
+        if token.validated is False:
+            return None # token not validated
+
```

Comment:
Yep - the URLSafeTimedSerializer is so that we can include the token in URLs (so that we can have someone click a link in an email) while the JSONSerializer allows us to specify the timeout at creation time rather than at validation time